### PR TITLE
Prompts: Add Tracks to Bloganuary nudges

### PR DESCRIPTION
**Pull request by dvdchr**
Depends on #22129 

As titled, here are the events that are tracked: 

Event | Event Name | Properties
-|-|-
Dashboard card shown | `my_site_dashboard_card_shown` | `type: bloganuary_nudge`, <br>`subtype: bloganuary_nudge`
Contextual menu tapped | `my_site_dashboard_contextual_menu_accessed` | `card: bloganuary_nudge`
Dashboard card hidden | `my_site_dashboard_card_hide_tapped` | `card: bloganuary_nudge`
Learn more button tapped | `bloganuary_nudge_my_site_card_learn_more_tapped` | `prompts_enabled: true, false`
Modal shown | `bloganuary_nudge_learn_more_modal_shown` | `prompts_enabled: true, false`
Modal button tapped | `bloganuary_nudge_learn_more_modal_action_tapped` | `action: turn_prompts_on, dismiss`
Modal dismissed | `bloganuary_nudge_learn_more_modal_dismissed` | n/a

## To test

- Launch the Jetpack app.
- 🔎 Verify that the `my_site_dashboard_card_shown` event is triggered.
- Tap 'Learn more'.
- 🔎 Verify that the `bloganuary_nudge_my_site_card_learn_more_tapped` event is triggered.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_shown` event is triggered. 
- Tap the Close button.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_dismissed` event is triggered.
- Tap the 'Learn more' button again, and tap the primary button on the modal.
- 🔎 Verify that the `bloganuary_nudge_learn_more_modal_action_tapped` event is triggered. 
- Tap the ellipsis button on the Bloganuary card.
- 🔎 Verify that the `my_site_dashboard_contextual_menu_accessed` event is triggered.
- Tap 'Hide this'. Note: Make sure to do this last (if you plan to verify other variations) since you'd have to reinstall the app to make it appear again.
- 🔎 Verify that the `my_site_dashboard_card_hide_tapped` event is triggered.

## Regression Notes
1. Potential unintended areas of impact
Should be none. Tracking events were added to new components and do not modify existing ones.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the changes.

3. What automated tests I added (or what prevented me from doing so)
N/A.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

UI Changes testing checklist:
N/A

- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Blog/Blogging Prompts/Bloganuary/BloganuaryOverlayViewController.swift [hunk 1]
```diff
@@ -75,6 +78,7 @@ class BloganuaryOverlayViewController: UIViewController {
 
         // Set up the close button in the navigation bar.
         let dismissAction = UIAction { [weak self] _ in
+            BloganuaryTracker.trackModalDismissed()
```
Comment (wargcm) [inline WordPress/Classes/ViewRelated/Blog/Blogging Prompts/Bloganuary/BloganuaryOverlayViewController.swift:81]: This function call is missing a parameter, causing the build to fail.

I'm going to modify the code on my end, assuming the local `promptsEnabled` bool is supposed to be passed here, and continue reviewing the rest of the PR.

**Review by wargcm — APPROVED** [review]
✅ `my_site_dashboard_card_shown`
✅ `my_site_dashboard_contextual_menu_accessed`
✅ `bloganuary_nudge_my_site_card_learn_more_tapped`
✅ `bloganuary_nudge_learn_more_modal_shown`
✅ `bloganuary_nudge_learn_more_modal_action_tapped`
✅ `bloganuary_nudge_learn_more_modal_dismissed`

There was just the missing parameter in that function call. LGTM! 🔥

File: WordPress/Classes/ViewRelated/Blog/Blogging Prompts/Bloganuary/BloganuaryOverlayViewController.swift (hunk 1 above)
Comment (dvdchr) [reply WordPress/Classes/ViewRelated/Blog/Blogging Prompts/Bloganuary/BloganuaryOverlayViewController.swift:81]: Thanks. This is fixed in https://github.com/wordpress-mobile/WordPress-iOS/pull/22130/commits/05356b066e8990b6998eac7dd69eb513b369b17a

**Comment by dvdchr** [comment]
I'm merging this ahead due to the same failures described in https://github.com/wordpress-mobile/WordPress-iOS/pull/22129#issuecomment-1836643483.

**Pull request merged**